src/screens/ModelPage/ShareModelBlock/EmbedPreview.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useRef } from 'react';
+import { useTheme } from '@material-ui/core/styles';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+
+import {
+  EmbedPreviewScrollContainer,
+  CenteredModal,
+  CodeSnippetPreview,
+  EmbedPreviewStyled,
+  EmbedDetailsStyled,
+  EmbedPreviewExitButton,
+} from './EmbedPreview.style';
+import CloseIcon from '@material-ui/icons/Close';
+import Grid from '@material-ui/core/Grid';
+import Snackbar from '@material-ui/core/Snackbar';
+import Typography from '@material-ui/core/Typography';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+
+import { EmbedButton } from './../ModelPage.style';
+
+import { useEmbed } from 'utils/hooks';
+
+export default function EmbedPreview({ open, onClose }) {
+  const {
+    EMBED_HEIGHT,
+    EMBED_WIDTH,
+    iFramePath,
+    jsCodeSnippet,
+    iFrameCodeSnippet,
+  } = useEmbed();
+
+  // https://material-ui.com/components/snackbars/#consecutive-snackbars
+  const queueRef = useRef([]);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [messageInfo, setMessageInfo] = useState(undefined);
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const processQueue = () => {
+    if (queueRef.current.length > 0) {
+      setMessageInfo(queueRef.current.shift());
+      setSnackbarOpen(true);
+    }
+  };
+
+  const handleCopySuccessMessage = message => () => {
+    queueRef.current.push({
+      message,
+      key: new Date().getTime(),
+    });
+
+    if (snackbarOpen) {
+      // immediately begin dismissing current message
+      // to start showing new one
+      setSnackbarOpen(false);
+    } else {
+      processQueue();
+    }
+  };
+
+  const handleClose = (event, reason) => {
```

Comment:
consider renaming these `handleSnackbarClose` and `handleSnackbarExited` to make it clearer they're for the snackbar, not the modal.